daemon/graphdriver/overlay2/overlay.go

```diff
@@ -538,6 +542,22 @@ func (d *Driver) Exists(id string) bool {
 
 // ApplyDiff applies the new layer into a root
 func (d *Driver) ApplyDiff(id string, parent string, diff io.Reader) (size int64, err error) {
+	if parent != "" {
+		lowers, err := d.getLowerDirs(id)
+		if err != nil {
+			return -1, err
+		}
+		parentDir := d.dir(parent)
+		var ld string
+		if len(lowers) > 0 {
+			ld = filepath.Dir(lowers[0])
+		}
+		if ld != parentDir {
```

Comment:
`ld` will be `<root>/<parentid>/diff` directory while `parentDir` will be `<root>/<parentid>`